src/api/app/controllers/webui/package_controller.rb

```diff
@@ -1129,4 +1138,19 @@ def check_build_log_access
 
     true
   end
+
+  def check_repository_and_architecture
```

Comment:
I agree, here is a card for this:
https://trello.com/c/DI5J8DyX/334-have-generic-way-of-handling-activerecord-notfound-errors-in-webui